File: Embodied-AI/ROS/catkin_ws/src/ssr_pkg/scripts/yao_node.py
#! /usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
if __name__=="__main__":
    rospy.init_node("yao_node")
    pub=rospy.Publisher("yao",String,queue_size=10)
    rate=rospy.Rate(10)
    while not rospy.is_shutdown():
        msg=String()
        msg.data="do this all day woman!!!"
        pub.publish(msg)
        rospy.loginfo("woman!!!")
        rate.sleep()

File: Embodied-AI/ROS/catkin_ws/src/ssr_pkg/src/yao_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
int main(int argc, char  *argv[])
{   
    ros::init(argc,argv,"yao_node");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::String>("yao",10);

    ros::Rate loop_rate(10);
    while(ros::ok())//使用ture时候运行节点不能响应外界信号
    {
        printf("keep\n");
        std_msgs::String msg;
        msg.data = "do this all day woman!!!";
        pub.publish(msg);
        loop_rate.sleep();
    }
    printf("hello world");
    return 0;
}
